### PR TITLE
week1-4 homework

### PR DESCRIPTION
**Pull request by yilanzh**

### CODE REVIEW

**Comment by yilanzh** [comment]
w1-5 homework

File: yilan_zhou/week_01/week1-4_geo_lab.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    } else {
+        console.log( "It is not obtuse.");
+    }
+return result;
+}
+
+isEquilateral(triangle);
+isIsosceles(triangle);
+calcTriangleArea(triangle);
+isObtuse(triangle);
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_geo_lab.js:105]: Good job Yilan!

File: yilan_zhou/week_01/week1-4_bank.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+function sumBankBalance(thisBank) {   
+    var total = 0;
+    for (var i = 0; i<thisBank.length;i++) {
+       total += thisBank[i].balance; 
+    }
+    return "Total Bank Balance: " + total;
+}
+
+function newBankAccount(thisBank,newName,newBalance) {
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_bank.js:24]: Good job, by taking bank as an argument you make your newBankAccount very flexible, so you can add accounts to different banks! :+1:

File: yilan_zhou/week_01/week1-4_bank.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    var newAccount = {
+    name: newName,
+    balance: newBalance
+};
+   thisBank.push(newAccount);
+   return "New account holder " + newName + " is added.";
+}
+
+function chgAccountBalance(thisBank,acName,acAmountChg) {
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_bank.js:33]: This function is a bit overcomplicated, it does few things, in future you might want to break this kind of functions to few smaller functions and just call on them inside of this one.

File: yilan_zhou/week_01/week1-4_bank.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            if ((thisBank[i].balance + acAmountChg)>=0) { //if withdraw balance <= a/c balance
+                thisBank[i].balance += acAmountChg;
+                return acName + "\'s balance has changed by " + acAmountChg + " to " + thisBank[i].balance;
+            } else {  //if withdraw amount > a/c balance
+            	return "Withdraw amount " + acAmountChg + " exceeds account balance of " + thisBank[i].balance + ", pls change amount.";
+            }
+            break;
+        }
+    }
+    if (!acFound) {  //if name is not found
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_bank.js:48]: That's good that you are making extra check for the account :)

File: yilan_zhou/week_01/week1-4_bank.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        return "Name not found";
+    }
+}
+
+function accountTransfer(thisBank, fromName,toName,amount) {
+	var fromID = -1;
+	var toID = -1;
+	//check if both parties are the same
+	if (fromName===toName) {
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_bank.js:57]: again, good to have extra check! Well done

File: yilan_zhou/week_01/week1-4_bank.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            }
+            break;
+        }
+    }
+    if (!acFound) {  //if name is not found
+        return "Name not found";
+    }
+}
+
+function accountTransfer(thisBank, fromName,toName,amount) {
```
Comment (ButenkoT) [inline yilan_zhou/week_01/week1-4_bank.js:53]: Same as previous function. Too many checks inside, you can take them out and make new functions, each of them should check for 1 thing, then just call them all inside of this function

File: yilan_zhou/week_01/w1-5_mta.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        arrStopName = getStopNames(endLine, connectStop, endStop);  //connect station to end stop
+        numStops += arrStopName.length;
+        result += "Take Line " +endLine + " " + connectStop + " and stop at "
+        + arrStopName.join(", ") + ", then get off at station " +
+        endStop + ". \nNumber of stops: " + numStops;
+    }
+    return "Instruction: " + result;
+}
+
+console.log (planTrip('6','Astor Place','N','Time Square'));
```
Comment (ButenkoT) [inline yilan_zhou/week_01/w1-5_mta.js:68]: Everything is good hear! Just all same, planTrip function is toooooo big, to improve code readability and flexibility you can break it on smaller function.

**Comment by ButenkoT** [comment]
Great job! All is very good and working proper :+1:  Just few notes in the code to improve it.

**Pull request merged**